workflow/controller/workflowpod.go

```diff
@@ -615,6 +615,14 @@ func isResourcesSpecified(ctr *apiv1.Container) bool {
 
 // addMetadata applies metadata specified in the template
 func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template, opts *createWorkflowPodOpts) {
+	// inherit workflow annotations and labels
+	for k, v := range woc.wf.ObjectMeta.Annotations {
+		pod.ObjectMeta.Annotations[k] = v
```

Comment:
I agree. We probably don't want to inherit directly as we don't know for sure whether to apply the same set of annotations/labels. Perhaps we could add something similar to podSpecPatch, e.g. podMetadataPatch.